testutil/sims/tx_helpers.go

```diff
@@ -131,19 +132,28 @@ func SignCheckDeliver(
 		Txs:    [][]byte{bz},
 	})
 
+	var finalizeSuccess bool
+	for _, result := range resBlock.TxResults {
+		if result.Code == 0 {
+			finalizeSuccess = true
+			break
+		}
+	}
```

Comment:
Since there is only a single tx and within that a single message, can we rather require len == 1 and that txresults[0].code == 0?